confdb/aclocal_datatype.m4

```diff
@@ -46,11 +46,11 @@ get_c_int_type() {
     pac_retval="unavailable"
 }
 
-# return the C type matching integer of size
+# return the C type matching float of size
 get_c_float_type() {
     len=$[]1
     pac_retval=
-    for c_type in float double __float128 "long_double" ; do
+    for c_type in float double _Float16 __float128 ; do
```

Comment:
There is a problem here. On i386/x86_64, the C "long double" type does not correspond to `__float128`, but to GCC's type `__float80`. However, note that `sizeof(__float80) == 16` as well as `sizeof(__float128) == 16`. Therefore a selection logic based on sizeof may not actually work.

To clarify: May concern is about `MPI_LONG_DOUBLE`. On i386/x86_64, it should map to the GCC type `__float80`, or what is called "original `long double`" in section 1.2 here: https://gcc.gnu.org/wiki/Ieee128PowerPC.

Reply:
I hard-coded `long double` here - https://github.com/pmodels/mpich/pull/7264/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#r1949556370

Reply:
Your link is not working on my end. However, if you are aware of the issue and MPICH will not eventually confuse "long double" with "__float128", then all good with me.

Reply:
https://github.com/pmodels/mpich/pull/7264#discussion_r1949556370

The link is to the comment below.